### PR TITLE
[script] [esp] configure channels via yaml, add recovery logic

### PR DESCRIPTION
**Pull request by KatoakDR**
### Background
* Following up on https://github.com/rpherbig/dr-scripts/pull/5046#pullrequestreview-693522835

### Changes
* Convert script to a class named `ESP`
* Read from YAML which channels to subscribe you to
* Add error recovery and handling logic if you're not wearing a gweth or it breaks
* Handle that `esp listen <channel>` toggles on/off you listening to that channel so the command may need to be run twice to get your preference configured

### Configuration
```yaml
# ESP Settings
# List the channels you want to subscribe to when `esp` script runs.
# You'll unsubscribe from any channels not listed.
esp_channels:
  - General
  - Trade
  - Guild
  - Race
  - Local
  - Private
  - Personal

# Which channel to set you up to send to by default.
esp_default_channel: General
```

### CODE REVIEW

**Comment by KatoakDR** [comment]
@robbintt Thanks for your earlier submision of the `esp` script. Here's my proposal to refactor it to read from YAML settings and such. Would like your review when you get a chance. Thanks

File: esp.lic [hunk 1]
```diff
@@ -1,13 +1,75 @@
...
+
+  def listen_to_these_channels_and_not_others(channels)
+    @all_channels.each do |channel|
+      listen = channels.include?(channel)
+      update_channel_subscription(channel, listen)
+    end
+  end
+
+  def update_channel_subscription(channel, listen)
+    # We don't have an easy way to know whether you are or are not currently
```
Comment (rpherbig) [inline esp.lic:38]: This feels like a good place for a UserVar

File: esp.lic [hunk 2]
```diff
@@ -1,13 +1,75 @@
...
+    when /Touch what?/
+      DRC.message("You don't have a gweth!")
+      exit
+    when /The chain snaps in two places and falls to the ground/
+      DRC.message("Uh oh, your gweth broke!")
+      exit
+    else
+      # Not wearing a gweth, try to wear one and retry
+      if retry_count < 1
+        DRCI.get_item_if_not_held?('gwethdesuan')
```
Comment (rpherbig) [inline esp.lic:65]: An opportunity for an early exit if this returns false?

File: esp.lic [hunk 3]
```diff
@@ -1,13 +1,75 @@
...
+        DRCI.wear_item?('gwethdesuan')
```
Comment (rpherbig) [inline esp.lic:66]: An opportunity for an early exit if this returns false?

File: esp.lic [hunk 4]
```diff
@@ -1,13 +1,75 @@
...
+      listening = Regexp.last_match[1] == 'open'
+      if listen != listening
+        update_channel_subscription(channel, listen)
+      end
+    else
+      DRC.message("Unrecognized channel: #{channel}")
+    end
+  end
+
+  def activate_gweth(retry_count = 0)
```
Comment (rpherbig) [inline esp.lic:52]: `activate_gweth?` maybe?

File: esp.lic [hunk 5]
```diff
@@ -1,13 +1,84 @@
...
+      if listen != listening
+        update_channel_subscription(channel, listen)
+      end
+    else
+      DRC.message("Unrecognized channel: #{channel}")
+    end
+  end
+
+  def activate_gweth?(retry_count = 0)
+    return false
```
Comment (rpherbig) [inline esp.lic:54]: Is this leftover test code?

File: esp.lic [hunk 6]
```diff
@@ -1,13 +1,84 @@
...
+      return true
+    when /Touch what?/
+      DRC.message("You don't have a gweth!")
+      exit
+    when /The chain snaps in two places and falls to the ground/
+      DRC.message("Uh oh, your gweth broke!")
+      exit
+    else
+      # Not wearing a gweth, try to wear one and retry
+      if retry_count < 1
```
Comment (rpherbig) [inline esp.lic:66]: If there is really only going to be a single retry, I'd prefer this to be a boolean

File: esp.lic (hunk 5 above)
Comment (KatoakDR) [reply esp.lic:54]: > Doesn't this negate the purpose of the function? Is this leftover dev/test code?

Doh! Residue from earlier testing (sigh)

File: esp.lic (hunk 6 above)
Comment (KatoakDR) [reply esp.lic]: changed to boolean

**Pull request merged**